k8s/upgrades/0.8.2-0.9.0/cstor/cstor-pool-update-082-090.yaml

```diff
@@ -0,0 +1,664 @@
+              - mountPath: /tmp
+                name: tmp
+              - mountPath: /var/openebs/sparse
+                name: sparse
+              - mountPath: /run/udev
+                name: udev
+  post: |-
+    {{- $message := printf "Successfully patched deployment {%s}." .UpgradeItem.name -}}
+    {{- $status := .Config.successStatus.value -}}
+    {{- $taskStatus := upgradeResultWithTaskStatus $status -}}
+    {{- $taskMessage := upgradeResultWithTaskMessage $message -}}
+    {{- $taskName := upgradeResultWithTaskName "upgrade-cstor-pool-082-090-patch-deployment-image" -}}
+    {{- $URName := upgradeResultWithTaskOwnerName .UpgradeItem.upgradeResultName -}}
+    {{- $URNamespace := upgradeResultWithTaskOwnerNamespace "default" -}}
+    {{- $taskStartTime := upgradeResultWithTaskStartTime now -}}
+    {{- $taskEndTime := upgradeResultWithTaskEndTime now -}}
+    {{- upgradeResultUpdateTasks $taskStartTime $URName $URNamespace $taskName $taskStatus $taskMessage $taskEndTime -}}
+
+---
+apiVersion: openebs.io/v1alpha1
+kind: RunTask
+metadata:
+  name: upgrade-cstor-pool-082-090-patch-deployment-image-status
+  namespace: default
+spec:
+  meta: |
+    {{ $isOldCStorPool := contains .TaskResult.poolDeployment.cstorpoolimage .Config.baseversion.value }}
+    {{ $isOldCStorPoolMGMT := eq .TaskResult.poolDeployment.cstorpoolmgmtimage "" }}
+    {{ $shouldpatch := or $isOldCStorPool $isOldCStorPoolMGMT | toString }}
+    id: patchDeploymentImageStatus
+    apiVersion: extensions/v1beta1
+    kind: Deployment
+    action: rolloutstatus
+    objectName: {{ .UpgradeItem.name }}
+    runNamespace: {{ .UpgradeItem.namespace }}
+    retry: "20,20s"
+  post: |
+    {{- jsonpath .JsonResult "{.isRolledout}" | trim | saveAs "patchDeploymentImageStatus.isRolledout" .TaskResult | noop -}}
+    {{- jsonpath .JsonResult "{.message}" | trim | saveAs "patchDeploymentImageStatus.rolloutStatus" .TaskResult | noop -}}
+    {{- $status := "" -}}
+    {{- if eq .TaskResult.patchDeploymentImageStatus.isRolledout  "true" }}
+    {{- $status = .Config.successStatus.value -}}
+    {{- else }}
+    {{- "waiting for deployment rollout" | saveAs "patchDeploymentImageStatus.verifyErr" .TaskResult | noop -}}
+    {{- $status = .Config.failStatus.value -}}
+    {{- end }}
+
+    {{- $message := printf "%s name: {%s} namespace: {%s}" .TaskResult.patchDeploymentImageStatus.rolloutStatus .UpgradeItem.name .UpgradeItem.namespace -}}
+
+
+    {{- $taskStatus := upgradeResultWithTaskStatus $status -}}
+    {{- $taskMessage := upgradeResultWithTaskMessage $message -}}
+    {{- $taskName := upgradeResultWithTaskName "upgrade-cstor-pool-082-090-patch-deployment-image-status" -}}
+    {{- $URName := upgradeResultWithTaskOwnerName .UpgradeItem.upgradeResultName -}}
+    {{- $URNamespace := upgradeResultWithTaskOwnerNamespace "default" -}}
+    {{- $taskStartTime := upgradeResultWithTaskStartTime now -}}
+    {{- $taskEndTime := upgradeResultWithTaskEndTime now -}}
```

Comment:
Ideal case, only endtime should get updated. if starttime is not set already then reuse the endtime. But we should not override the starttime